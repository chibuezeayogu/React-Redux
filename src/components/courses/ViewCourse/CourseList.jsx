import React from 'react';
import { object, func } from 'prop-types';
import { Link } from 'react-router-dom';

const CourseList = ({ course, showDeleteModal }) => (
	<tr>
		<td scope="row"><a href={course.watchHref} target="_blank">Watch</a></td>
		<td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
		<td>{course.authorId}</td>
		<td>{course.category}</td>
		<td>{course.length}</td>
		<td>
			<span
				type="button"
				tabIndex={1}
				onClick={() => showDeleteModal(course)}
			>
			delete
			</span>
		</td>
	</tr>
);

CourseList.propTypes = {
	course: object,
	showDeleteModal: func
};

export default CourseList;
