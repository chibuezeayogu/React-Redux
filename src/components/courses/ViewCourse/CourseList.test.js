/* eslint-disable max-len */

import React from 'react';
import { shallow } from 'enzyme';

import CourseList from './CourseList';

describe('CourseList Component', () => {
	const props = {
		courses: [{
			id: 'react-flux-building-applications',
			title: 'Building Applications in React and Flux',
			watchHref: 'http://www.pluralsight.com/courses/react-flux-building-applications',
			authorId: 'cory-house',
			length: '5:08',
			category: 'JavaScript'
		}]
	};
  
	const wrapper = shallow(<CourseList {...props} />);
  
	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});