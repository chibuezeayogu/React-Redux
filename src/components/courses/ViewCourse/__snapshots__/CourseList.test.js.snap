// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList Component should render correctly 1`] = `
<tr>
  <td
    scope="row"
  >
    <a
      href="http://www.pluralsight.com/courses/react-flux-building-applications"
      target="_blank"
    >
      Watch
    </a>
  </td>
  <td>
    <Link
      replace={false}
      to="/course/react-flux-building-applications"
    >
      Building Applications in React and Flux
    </Link>
  </td>
  <td>
    cory-house
  </td>
  <td>
    JavaScript
  </td>
  <td>
    5:08
  </td>
  <td>
    <span
      onClick={[Function]}
      tabIndex={1}
      type="button"
    >
      delete
    </span>
  </td>
</tr>
`;
